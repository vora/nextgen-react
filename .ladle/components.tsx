import type { GlobalProvider } from "@ladle/react";
import React from "react";
import { LadleErrorBoundary } from "./LadleErrorBoundary";
import "../src/styles.css";

export const Provider: GlobalProvider = ({ children }) => (
  <React.StrictMode>
    <LadleErrorBoundary>{children}</LadleErrorBoundary>
  </React.StrictMode>
);
