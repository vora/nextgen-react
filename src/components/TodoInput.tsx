import { useState } from "react";

const PlusIcon = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    className="h-6 w-6"
    fill="none"
    viewBox="0 0 24 24"
    stroke="currentColor"
    strokeWidth={2}
  >
    <path strokeLinecap="round" strokeLinejoin="round" d="M12 4v16m8-8H4" />
  </svg>
);

interface Props {
  onAdd?: (todo: string) => void;
}

export const TodoInput: React.FC<Props> = ({ onAdd }) => {
  const [value, setValue] = useState<string>("");

  const handleAdd = () => onAdd && onAdd(value);

  return (
    <div className="flex flex-col">
      <div className="flex flex-row">
        <input
          value={value}
          onChange={(e) => setValue(e.target.value)}
          placeholder="Add a TODO"
          className="p-4 rounded my-1 text-lg w-full"
        />
        <button
          onClick={handleAdd}
          className="border p-4 m-2 rounded shadow bg-white hover:bg-gray-200 active:bg-gray-400 flex flex-row items-center"
        >
          <PlusIcon />
        </button>
      </div>
    </div>
  );
};
