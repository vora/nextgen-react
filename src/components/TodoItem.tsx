export const TodoItem = () => {
  return <p>TODO Item</p>;
};
