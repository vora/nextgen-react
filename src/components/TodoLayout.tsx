import { ReactNode } from "react";

interface Props {
  children: ReactNode;
}

export const TodoLayout: React.FC<Props> = ({ children }) => {
  return (
    <div className="h-full w-full">
      <div className="border max-w-2xl rounded-lg shadow">
        <div className="bg-blue-500 p-4 text-center rounded-t-lg">
          <h1 className="text-white text-2xl font-bold">TODO List</h1>
        </div>
        <div className="p-4">{children}</div>
      </div>
    </div>
  );
};
