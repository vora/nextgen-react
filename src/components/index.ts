export * from "./TodoInput";
export * from "./TodoItem";
export * from "./TodoLayout";
