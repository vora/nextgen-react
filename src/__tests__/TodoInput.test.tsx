import renderer from "react-test-renderer";
import { TodoInput } from "~/components";

test("TodoInput", () => {
  const component = renderer.create(<TodoInput />);

  expect(component.toJSON()).toMatchSnapshot();
});
