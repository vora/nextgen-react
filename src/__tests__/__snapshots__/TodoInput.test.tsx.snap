// Vitest Snapshot v1

exports[`TodoInput 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="flex flex-row"
  >
    <input
      className="p-4 rounded my-1 text-lg w-full"
      onChange={[Function]}
      placeholder="Add a TODO"
      value=""
    />
    <button
      className="border p-4 m-2 rounded shadow bg-white hover:bg-gray-200 active:bg-gray-400 flex flex-row items-center"
      onClick={[Function]}
    >
      <svg
        className="h-6 w-6"
        fill="none"
        stroke="currentColor"
        strokeWidth={2}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4v16m8-8H4"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
